feat: add `blas/ext/base/sindex-of`

headlessNode opened the pull request:
---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes. report:
  - task: lint_filenames status: passed
  - task: lint_editorconfig status: passed
  - task: lint_markdown status: passed
  - task: lint_package_json status: passed
  - task: lint_repl_help status: passed
  - task: lint_javascript_src status: passed
  - task: lint_javascript_cli status: na
  - task: lint_javascript_examples status: passed
  - task: lint_javascript_tests status: passed
  - task: lint_javascript_benchmarks status: passed
  - task: lint_python status: na
  - task: lint_r status: na
  - task: lint_c_src status: na
  - task: lint_c_examples status: na
  - task: lint_c_benchmarks status: na
  - task: lint_c_tests_fixtures status: na
  - task: lint_shell status: na
  - task: lint_typescript_declarations status: passed
  - task: lint_typescript_tests status: passed
  - task: lint_license_headers status: passed ---

Progresses #2656 .

## Description

> What is the purpose of this pull request?

This pull request:

-   add `blas/ext/base/sindexof`

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   Progresses #2656  

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

headlessNode committed:
feat: add index-of

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

headlessNode committed:
fix: function name

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/sindex-of/README.md:
````diff
@@ -0,0 +1,170 @@
+var sindexOf = require( '@stdlib/blas/ext/base/sindex-of' );
+```
+
+#### sindexOf( N, searchElement, x, strideX\[, offsetX ] )
````
@headlessNode This is not correct. Our practice is to **always** have separate APIs. One for providing an offset and another in which the offset is computed. This follows BLAS conventions. I am not sure why you decided to make the API variadic.

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/sindex-of/README.md:
```diff
@@ -0,0 +1,170 @@
+-   **searchElement**: element to search the index of.
+-   **x**: input [`Float32Array`][@stdlib/array/float32].
+-   **strideX**: index increment.
+-   **offsetX**: starting index (_optional_).
```
Incorrect. Please follow existing practice.

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/sindex-of/docs/repl.txt:
```diff
@@ -0,0 +1,40 @@
+
+{{alias}}( N, searchElement, x, strideX[, offsetX ] )
```
Incorrect.

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/sindex-of/docs/types/index.d.ts:
```diff
@@ -0,0 +1,48 @@
+* var idx = sindexOf( x.length, 2.0, x, 1 );
+* // returns 1
+*/
+declare function sindexOf( N: number, searchElement: number, x: Float32Array, strideX: number, offsetX?: number ): number;
```
In correct.

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/sindex-of/docs/types/index.d.ts:
```diff
@@ -0,0 +1,48 @@
+* ## Notes
+*
+* -   If unable to find an element which equals a provided search element, the function returns `-1`.
+* -   If `offsetX` is less than zero, the starting index is resolved relative to the last array element, with the last array element corresponding to `offsetX = -1`.
```
Offsets are never negative. Why are you designing the API this way? Please show me what BLAS package supports this behavior.

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/sindex-of/lib/main.js:
```diff
@@ -0,0 +1,76 @@
+* @param {Number} searchElement - number to search index of
+* @param {Float32Array} x - input array
+* @param {integer} strideX - index increment
+* @param {NonNegativeInteger} [offsetX] - starting index
```
`NonNegativeInteger` directly contradicts what you stated in the `index.d.ts` file.

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/sindex-of/lib/main.js:
```diff
@@ -0,0 +1,76 @@
+		return -1;
+	}
+
+	if ( offsetX === void 0 ) {
```
Incorrect.

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/sindex-of/lib/main.js:
```diff
@@ -0,0 +1,76 @@
+
+	if ( offsetX === void 0 ) {
+		if ( strideX < 0 ) {
+			ix = x.length - 1;
```
We should NOT be assuming or inquiring anything about `x` length. Plus, this doesn't make sense at all. `N` and `stride` completely specify the array view defining what values are indexed.

headlessNode committed:
fix: apply suggestions from code review

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

headlessNode commented on lib/node_modules/@stdlib/blas/ext/base/sindex-of/lib/ndarray.js:
```diff
@@ -0,0 +1,62 @@
+// MAIN //
+
+/**
+* Returns the index of a specified search element in a single-precision floating-point strided array.
```
```suggestion
* Returns the index of a specified search element in a single-precision floating-point strided array using alternative indexing semantics.
```

headlessNode committed:
chore: apply code review suggestion

Signed-off-by: Muhammad Haris <101793258+headlessNode@users.noreply.github.com>

kgryte commented:
@headlessNode Would you mind going ahead and adding the C implementation and associated benchmarks, examples, docs, and tests?

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/sindex-of/README.md:
```diff
@@ -0,0 +1,196 @@
+The function has the following parameters:
+
+-   **N**: number of indexed elements.
+-   **searchElement**: element to search the index of.
```
```suggestion
-   **searchElement**: search element.
```

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/sindex-of/README.md:
```diff
@@ -0,0 +1,196 @@
+-   **N**: number of indexed elements.
+-   **searchElement**: element to search the index of.
+-   **x**: input [`Float32Array`][@stdlib/array/float32].
+-   **strideX**: index increment.
```
```suggestion
-   **strideX**: stride length.
```

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/sindex-of/README.md:
```diff
@@ -0,0 +1,196 @@
+-   **x**: input [`Float32Array`][@stdlib/array/float32].
+-   **strideX**: index increment.
+
+If the function is unable to find an element which equals a provided search element, the function returns `-1`.
```
```suggestion
If the function is unable to find a search element, the function returns `-1`.
```

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/sindex-of/README.md:
```diff
@@ -0,0 +1,196 @@
+var x = new Float32Array( [ -2.0, 1.0, 3.0, -5.0, 4.0, 0.0, -1.0, 3.0 ] );
+
+var idx = sindexOf( 4, -1.0, x, 2 );
+// returns 6
```
This is incorrect. `N` specifies the view length. Here, that "view" is four elements. `6` is the index in the underlying buffer. That is not what we care about in this case.

kgryte replied:
See also [`blas/base/isamax`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/blas/base/isamax).

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/sindex-of/README.md:
```diff
@@ -0,0 +1,196 @@
+
+// Find index...
+var idx = sindexOf( 3, -6.0, x1, 2 );
+// returns 4
```
Same thing. This return value is incorrect.

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/sindex-of/README.md:
```diff
@@ -0,0 +1,196 @@
+var x = new Float32Array( [ -2.0, 1.0, 3.0, -5.0, 4.0, 0.0, -1.0, 3.0 ] );
+
+var idx = sindexOf.ndarray( 3, 3.0, x, 1, x.length-3 );
+// returns 7
```
This is incorrect.

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/sindex-of/benchmark/benchmark.length.js:
```diff
@@ -0,0 +1,96 @@
+
+		b.tic();
+		for ( i = 0; i < b.iterations; i++ ) {
+			out = sindexOf( x.length, 2.0, x, 1 );
```
```suggestion
			out = sindexOf( x.length, len+1, x, 1 );
```

`2.0` obviously does not make sense here, as then the benchmarks are effectively constant time. You should be benchmarking the worst case scenario (i.e., when you need to search an entire array).

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/sindex-of/benchmark/benchmark.length.js:
```diff
@@ -0,0 +1,96 @@
+		len = pow( 10, i );
+
+		f = createBenchmark( len );
+		bench( pkg+':dtype=float32,len='+len, f );
```
Why are we including `dtype` here? Only one dtype is allowed/expected.

```suggestion
		bench( pkg+':len='+len, f );
```

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/sindex-of/benchmark/benchmark.ndarray.js:
```diff
@@ -0,0 +1,96 @@
+
+		b.tic();
+		for ( i = 0; i < b.iterations; i++ ) {
+			out = sindexOf( x.length, 2.0, x, 1, 0 );
```
```suggestion
			out = sindexOf( x.length, len+1, x, 1, 0 );
```

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/sindex-of/benchmark/benchmark.ndarray.js:
```diff
@@ -0,0 +1,96 @@
+		len = pow( 10, i );
+
+		f = createBenchmark( len );
+		bench( pkg+':dtype=float32,len='+len, f );
```
```suggestion
		bench( pkg+':len='+len, f );
```

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/sindex-of/lib/main.js:
```diff
@@ -0,0 +1,53 @@
+* Returns the index of a specified search element in a single-precision floating-point strided array.
+*
+* @param {PositiveInteger} N - number of indexed elements
+* @param {Number} searchElement - number to search index of
```
```suggestion
* @param {number} searchElement - search element
```

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/sindex-of/lib/main.js:
```diff
@@ -0,0 +1,53 @@
+* @param {PositiveInteger} N - number of indexed elements
+* @param {Number} searchElement - number to search index of
+* @param {Float32Array} x - input array
+* @param {integer} strideX - index increment
```
```suggestion
* @param {integer} strideX - stride length
```

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/sindex-of/lib/ndarray.js:
```diff
@@ -0,0 +1,62 @@
+* Returns the index of a specified search element in a single-precision floating-point strided array using alternative indexing semantics.
+*
+* @param {PositiveInteger} N - number of indexed elements
+* @param {Number} searchElement - number to search index of
```
```suggestion
* @param {number} searchElement - search element
```

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/sindex-of/lib/ndarray.js:
```diff
@@ -0,0 +1,62 @@
+* @param {PositiveInteger} N - number of indexed elements
+* @param {Number} searchElement - number to search index of
+* @param {Float32Array} x - input array
+* @param {integer} strideX - index increment
```
```suggestion
* @param {integer} strideX - stride length
```

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/sindex-of/lib/ndarray.js:
```diff
@@ -0,0 +1,62 @@
+	ix = offsetX;
+	for ( i = 0; i < N; i++ ) {
+		if ( x[ ix ] === searchElement ) {
+			return ix;
```
```suggestion
			return i;
```

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/sindex-of/test/test.main.js:
```diff
@@ -0,0 +1,96 @@
+	t.strictEqual( actual, 3, 'returns expected value' );
+
+	actual = sindexOf( x.length, 3.0, x, -1 );
+	t.strictEqual( actual, 5, 'returns expected value' );
```
This is incorrect.

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/sindex-of/test/test.main.js:
```diff
@@ -0,0 +1,96 @@
+	t.strictEqual( actual, 1, 'returns expected value' );
+
+	actual = sindexOf( x.length, 2.0, x, -1 );
+	t.strictEqual( actual, 3, 'returns expected value' );
```
This is incorrect.

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/sindex-of/test/test.main.js:
```diff
@@ -0,0 +1,96 @@
+
+	// Negative stride...
+	actual = sindexOf( x.length, 1.0, x, -1 );
+	t.strictEqual( actual, 1, 'returns expected value' );
```
This is incorrect.

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/sindex-of/test/test.ndarray.js:
```diff
@@ -0,0 +1,102 @@
+	t.strictEqual( actual, 4, 'returns expected value' );
+
+	// Negative stride...
+	actual = sindexOf( x.length, 1.0, x, -1, x.length-1 );
+	t.strictEqual( actual, 1, 'returns expected value' );
+
+	actual = sindexOf( x.length, 2.0, x, -2, x.length-1 );
+	t.strictEqual( actual, 3, 'returns expected value' );
+
+	actual = sindexOf( x.length, 1.0, x, -2, x.length-2 );
+	t.strictEqual( actual, 0, 'returns expected value' );
```
These are all incorrect.

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/sindex-of/test/test.ndarray.js:
```diff
@@ -0,0 +1,102 @@
+	t.strictEqual( actual, 0, 'returns expected value' );
+
+	actual = sindexOf( x.length, 2.0, x, 1, 1 );
+	t.strictEqual( actual, 2, 'returns expected value' );
```
Incorrect.

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/sindex-of/test/test.ndarray.js:
```diff
@@ -0,0 +1,102 @@
+	t.strictEqual( actual, 2, 'returns expected value' );
+
+	actual = sindexOf( x.length, 3.0, x, 1, 2 );
+	t.strictEqual( actual, 4, 'returns expected value' );
```
Incorrect.

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/sindex-of/lib/ndarray.js:
```diff
@@ -0,0 +1,62 @@
+	var ix;
+	var i;
+
+	if ( N <= 0 || x.length === 0 ) {
```
```suggestion
	if ( N <= 0 ) {
```

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/sindex-of/test/test.ndarray.js:
```diff
@@ -0,0 +1,102 @@
+	t.end();
+});
+
+tape( 'the function returns `-1` if provided an offset which exceeds the maximum array index', function test( t ) {
+	var actual;
+
+	actual = sindexOf( 3, 2.0, new Float32Array( [ 1.0, 2.0, 3.0 ] ), 1, 20 );
+	t.strictEqual( actual, -1, 'returns expected value' );
+
+	t.end();
+});
```
This test should be removed. In strided array implementations, we don't perform buffer bounds checking.

kgryte replied:
Again, it would be useful to refer to https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/blas/base/isamax/lib/ndarray.js.

headlessNode committed:
fix: apply suggestions from code review

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

headlessNode committed:
Merge branch 'blas-sindex-of' of https://github.com/headlessNode/stdlib into blas-sindex-of

headlessNode committed:
feat: add c apis

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: missing_dependencies
  - task: lint_c_examples
    status: missing_dependencies
  - task: lint_c_benchmarks
    status: missing_dependencies
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

headlessNode commented on lib/node_modules/@stdlib/blas/ext/base/sindex-of/README.md:
````diff
@@ -0,0 +1,315 @@
+### Usage
+
+```c
+#include "stdlib/blas/ext/base/sindexOf.h"
````
```suggestion
#include "stdlib/blas/ext/base/sindexof.h"
```

headlessNode commented on lib/node_modules/@stdlib/blas/ext/base/sindex-of/README.md:
````diff
@@ -0,0 +1,315 @@
+### Examples
+
+```c
+#include "stdlib/blas/ext/base/sindexOf.h"
````
```suggestion
#include "stdlib/blas/ext/base/sindexof.h"
```

headlessNode commented on lib/node_modules/@stdlib/blas/ext/base/sindex-of/examples/c/example.c:
```diff
@@ -0,0 +1,37 @@
+	// Specify a stride:
+	const int strideX = 1;
+
+	// Fill the array:
```
```suggestion
	// Search the index:
```

headlessNode commented on lib/node_modules/@stdlib/blas/ext/base/sindex-of/include/stdlib/blas/ext/base/sindexof.h:
```diff
@@ -0,0 +1,45 @@
+* limitations under the License.
+*/
+
+#ifndef STDLIB_BLAS_EXT_BASE_SNANSUMKBN_H
+#define STDLIB_BLAS_EXT_BASE_SNANSUMKBN_H
```
```suggestion
#ifndef STDLIB_BLAS_EXT_BASE_SINDEXOF_H
#define STDLIB_BLAS_EXT_BASE_SINDEXOF_H
```

headlessNode committed:
chore: apply suggestions from code review

Signed-off-by: Muhammad Haris <101793258+headlessNode@users.noreply.github.com>

headlessNode commented on lib/node_modules/@stdlib/blas/ext/base/sindex-of/src/addon.c:
```diff
@@ -0,0 +1,63 @@
+	STDLIB_NAPI_ARGV_INT64( env, searchElement, argv, 1 );
+	STDLIB_NAPI_ARGV_INT64( env, strideX, argv, 3 );
+	STDLIB_NAPI_ARGV_STRIDED_FLOAT32ARRAY( env, X, N, strideX, argv, 2 );
+	STDLIB_NAPI_CREATE_INT32( env, API_SUFFIX(stdlib_strided_sindexOf)( N, searchElement, X, strideX ), idx ); // TODO: Need to revisit in order to support int64_t for CBLAS_INT
```
@kgryte  Is this okay?

headlessNode commented on lib/node_modules/@stdlib/blas/ext/base/sindex-of/examples/c/example.c:
```diff
@@ -0,0 +1,37 @@
+
+int main( void ) {
+	// Create a strided array:
+	float x[] = { 1.0f, -2.0f, 3.0f, -4.0f, 5.0f, -6.0f, 7.0f, -8.0f };
```
```suggestion
	const float x[] = { 1.0f, -2.0f, 3.0f, -4.0f, 5.0f, -6.0f, 7.0f, -8.0f };
```

headlessNode committed:
fix: lint error

Signed-off-by: Muhammad Haris <101793258+headlessNode@users.noreply.github.com>

headlessNode commented on lib/node_modules/@stdlib/blas/ext/base/sindex-of/benchmark/c/benchmark.length.c:
```diff
@@ -0,0 +1,191 @@
+	idx = -1;
+	t = tic();
+	for ( i = 0; i < iterations; i++ ) {
+		idx = stdlib_strided_sindexOf( len, 1000.0f, x, 1 );
```
```suggestion
		// cppcheck-suppress uninitvar
		idx = stdlib_strided_sindexOf( len, 1000.0f, x, 1 );
```

headlessNode commented on lib/node_modules/@stdlib/blas/ext/base/sindex-of/benchmark/c/benchmark.length.c:
```diff
@@ -0,0 +1,191 @@
+	idx = -1;
+	t = tic();
+	for ( i = 0; i < iterations; i++ ) {
+		idx = stdlib_strided_sindexOf_ndarray( len, 1000.0f, x, 1, 0 );
```
```suggestion
		// cppcheck-suppress uninitvar
		idx = stdlib_strided_sindexOf_ndarray( len, 1000.0f, x, 1, 0 );
```

headlessNode committed:
chore: ignore uninitvar

Signed-off-by: Muhammad Haris <101793258+headlessNode@users.noreply.github.com>

headlessNode commented on lib/node_modules/@stdlib/blas/ext/base/sindex-of/benchmark/c/benchmark.length.c:
```diff
@@ -0,0 +1,193 @@
+	t = tic();
+	for ( i = 0; i < iterations; i++ ) {
+		// cppcheck-suppress uninitvar
+		idx = stdlib_strided_sindexOf( len, 1000.0f, x, 1 );
```
Here and below, I think we can't use `searchElement` as in JS benchmarks. I think we could use `-1` since that won't be present in the array, hence going over all the elements i.e. worst case scenario

kgryte replied:
Theoretically, `-1` is possible based on L104. Instead, use `20000.0f` (i.e., a value well outside the range of L105.

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/sindex-of/benchmark/c/benchmark.length.c:
```diff
@@ -0,0 +1,193 @@
+	t = tic();
+	for ( i = 0; i < iterations; i++ ) {
+		// cppcheck-suppress uninitvar
+		idx = stdlib_strided_sindexOf_ndarray( len, 1000.0f, x, 1, 0 );
```
Same comment.

headlessNode committed:
refactor: c benchmark

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: missing_dependencies
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

headlessNode committed:
Merge branch 'blas-sindex-of' of https://github.com/headlessNode/stdlib into blas-sindex-of

headlessNode committed:
chore: lint error

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: missing_dependencies
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte committed:
chore: clean-up

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: passed
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

kgryte reviewed (approved):
LGTM after a fair amount of clean-up.

headlessNode commented on lib/node_modules/@stdlib/blas/ext/base/sindex-of/include/stdlib/blas/ext/base/sindex_of.h:
```diff
@@ -29,17 +29,17 @@ extern "C" {
+* Returns the first index of a specified search element in a single-precision floating-point strided array.
 */
-CBLAS_INT API_SUFFIX(stdlib_strided_sindexOf)( const CBLAS_INT N, const CBLAS_INT searchElement, const float *X, const CBLAS_INT strideX );
+CBLAS_INT API_SUFFIX(stdlib_strided_sindex_of)( const CBLAS_INT N, const CBLAS_INT searchElement, const float *X, const CBLAS_INT strideX );
```
@kgryte `searchElement` should be `float`, no?

kgryte replied:
Yes, that is correct.

headlessNode replied:
Opened the fix PR.
